core/trino-main/src/main/java/io/trino/execution/TaskStatus.java

```diff
@@ -66,7 +66,7 @@ public class TaskStatus
     private final DataSize memoryReservation;
     private final DataSize peakMemoryReservation;
     private final DataSize revocableMemoryReservation;
-
+    private final DataSize queryMemoryReservation;
```

Comment:
It looks odd to have a query level metric inside task status.
Could we calculate this from existing task status by grouping `memoryReservation` by `queryId` and `nodeId` ?